core/creators/inline.js

```diff
@@ -93,11 +93,16 @@
 
 		// Handle editor destroying.
 		editor.on( 'destroy', function() {
+			var container = editor.container;
 			// Remove container from DOM if inline-textarea editor.
 			// Show <textarea> back again.
+			// Editor can be destroyed before container is created (#3115).
+			if ( textarea && container ) {
```

Comment:
It can, but we are using `isDetached()` condition to prevent further initialization. Here we are destroying editor, I see no reason to alter this behaviour when editor is detached.

Reply:
> I'm not sure if I understand. Clearing custom data / removing editor from DOM won't cause any issues when it's detached? If so, additional check if a container is detached may be redundant.

Custom data is stored in one `customData` object, it's not accessible from global scope, but it has only one instance. Each element which has custom data has it's unique number stored as html attribute which is also key for entry in `customData. We can read html attribute of detached elements, so no issue here.
Also calling `element.remove()` on detached element doesn't seem to cause any issue.

>Also, should custom data by cleared only for textarea inline editor? It shouldn't be removed obviously, but we probably should remove our event listeners, etc. On the other hand, not sure if we should introduce such changes with this PR 🤔

In case of non-textarea based inline editor 
```editor.container.equals( editor.element ) === true```

So everything is cleaned here:
https://github.com/ckeditor/ckeditor-dev/blob/0d2458b99b811b5f235e0e1620f32aa22eca27b7/core/creators/inline.js#L109